ProcessMaker/processmaker
# Enhance Script Executor: Implement 'is_system' Attribute & Hide System Resources

## Issue & Reproduction Steps
This PR enhances the management of Script Executors by enabling the capability to hide certain script executors from the user interface. 

## Solution

- Implemented a migration to add an `is_system` column to the Script Executors table.
- Integrated the HideSystemResourcesTrait on the ScriptExecutor Model to facilitate resource concealment.
- Updated Script Executor queries to exclusively retrieve nonSystem records.


## How to Test

1. Ensure that the migration is executed.
2. Install the CDATA or Node SSR executors using the branch `task/FOUR-11143`
3. Go to the 'Admin' section and click on 'Script Executors'.
4. Verify that both CDATA and Node SSR executors are not listed.
5. Go to the 'Designer' section and select 'Scripts'.
6. Create a new script.
7. Verify that within the 'Create New Script' model, the options for Languages do not include 'CDATA' or 'Node SSR'.

## Related Tickets & Packages
- Ticket [FOUR-11143](https://processmaker.atlassian.net/browse/FOUR-11143)
- [CDATA Executor PR](https://github.com/ProcessMaker/docker-executor-node-ssr/pull/10)
- [Node SST Executor PR](https://github.com/ProcessMaker/docker-executor-cdata/pull/6)

ci:next
ci:docker-executor-cdata:task/FOUR-11143
ci:docker-executor-node-ssr:task/FOUR-11143

## Code Review Checklist


[FOUR-11143]: https://processmaker.atlassian.net/browse/FOUR-11143?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
